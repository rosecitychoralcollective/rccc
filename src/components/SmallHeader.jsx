import React, { useState } from 'react';
import {
  AppBar, IconButton, MenuItem, Menu, Popover, Toolbar,
} from '@material-ui/core';
import FacebookIcon from '@material-ui/icons/Facebook';
import EmailIcon from '@material-ui/icons/Email';
import PhoneIcon from '@material-ui/icons/Phone';
import ContactSupportIcon from '@material-ui/icons/ContactSupport';
import MenuIcon from '@material-ui/icons/Menu';
import _ from 'lodash';
import translateString from '../utils/StringHelper';

const t = translateString;

const SmallHeader = ({ props }) => {
  const {
    classes,
    tabs,
    changeTabs,
    handleFbClick,
    handleEmailClick,
    handlePhoneClick,
    handlePopperClose,
    popperAnchorEl,
  } = props;
  const [contactAnchorEl, setContactAnchorEl] = useState(null);
  const [tabAnchorEl, setTabAnchorEl] = useState(null);

  const handleContactClose = () => {
    setContactAnchorEl(null);
  };

  const handleContactClick = (event) => {
    setContactAnchorEl(event.currentTarget);
  };

  const handleTabClose = () => {
    setTabAnchorEl(null);
  };

  const handleTabClick = (index) => {
    changeTabs('', index);
    handleTabClose();
  };

  const openTabMenu = (event) => {
    setTabAnchorEl(event.currentTarget);
  };

  const handleMenuClick = (str, evt) => {
    if (str === 'fb') {
      handleFbClick();
    } else if (str === 'email') {
      handleEmailClick();
    } else {
      handlePhoneClick(evt);
    }
    handleContactClose();
  };

  return (
    <AppBar
      className={classes.header}
    >
      <Toolbar>
        <div className={classes.tabs}>
          <IconButton
            onClick={openTabMenu}
            color="inherit"
          >
            <MenuIcon />
          </IconButton>
          <Menu
            id="nav-menu"
            anchorEl={tabAnchorEl}
            keepMounted
            open={Boolean(tabAnchorEl)}
            onClose={handleTabClose}
          >
            {tabs.map((tab, index) => (
              <MenuItem
                key={tab.name}
                onClick={() => handleTabClick(index)}
              >
                {_.startCase(tab.name)}
              </MenuItem>
            ))}
          </Menu>
        </div>
        <div className={classes.contactButtons}>
          <IconButton
            onClick={handleContactClick}
            color="inherit"
          >
            <ContactSupportIcon />
          </IconButton>
          <Menu
            id="contact-menu"
            keepMounted
            anchorEl={contactAnchorEl}
            open={Boolean(contactAnchorEl)}
            onClose={handleContactClose}
          >
            <MenuItem
              key="fb"
              onClick={() => handleMenuClick('fb')}
              id="facebook-button"
            >
              <FacebookIcon />
            </MenuItem>
            <MenuItem
              key="email"
              onClick={() => handleMenuClick('email')}
              id="email-button"
            >
              <EmailIcon />
            </MenuItem>
            <MenuItem
              key="phone"
              onClick={(evt) => handleMenuClick('phone', evt)}
              id="phone-button"
            >
              <PhoneIcon />
            </MenuItem>
          </Menu>
          <Popover
            open={Boolean(popperAnchorEl)}
            anchorEl={popperAnchorEl}
            onClose={handlePopperClose}
            anchorOrigin={{
              vertical: 'bottom',
              horizontal: 'center',
            }}
            transformOrigin={{
              vertical: 'top',
              horizontal: 'center',
            }}
          >
            <div id="phone-number" className={classes.phonePopover}>{t('Phone-Number')}</div>
          </Popover>
        </div>
      </Toolbar>
    </AppBar>
  );
};

export default SmallHeader;
