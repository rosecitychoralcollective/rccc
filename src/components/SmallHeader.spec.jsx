import React from 'react';
import { shallow } from 'enzyme';
import { Menu, MenuItem, Tab, Tabs } from '@material-ui/core';
import SmallHeader from './SmallHeader';
import translateString from '../utils/StringHelper';

const t = translateString;

const tabs = [{ name: 'asdf', route: '/asdf' }, { name: 'fdsa', route: '/fdsa' }];
const tabIndex = 0;
const changeTabs = jest.fn();
const popperAnchorEl = null;
const classes = {};
const setStyle = jest.fn();
const handleFbClick = jest.fn();
const handleEmailClick = jest.fn();
const handlePhoneClick = jest.fn();
const props = {
  tabs,
  tabIndex,
  changeTabs,
  popperAnchorEl,
  classes,
  setStyle,
  handleFbClick,
  handleEmailClick,
  handlePhoneClick,
};
const OLD_ENV = process.env.NODE_ENV;

describe('Full header component', () => {
  let component;
  beforeEach(() => {
    jest.resetModules();
    process.env.NODE_ENV = OLD_ENV;
    component = shallow(<SmallHeader props={props} />);
  });

  afterEach(() => {
    process.env.NODE_ENV = OLD_ENV;
    handleFbClick.mockReset();
    handleEmailClick.mockReset();
    handlePhoneClick.mockReset();
  });

  it('Renders without crashing', () => {
    expect(component.html()).toMatchSnapshot();
  });
  it('Does not render the style menu', () => {
    expect(component.exists('#style-menu')).not.toBe(true);
  });
  it('Does not render any tabs', () => {
    expect(component.find(Tab)).toHaveLength(0);
    expect(component.find(Tabs)).toHaveLength(0);
  });
  it('Renders two Menu components', () => {
    expect(component.find(Menu)).toHaveLength(2);
  });
  it('Has a nav menu component', () => {
    expect(component.exists('#nav-menu')).toBe(true);
  })
  it('Has a contact menu component', () => {
    expect(component.exists('#contact-menu')).toBe(true);
  })
  it('Clicks the facebook button and no other buttons', () => {
    component.find('#facebook-button').simulate('click');
    expect(handleFbClick).toHaveBeenCalled();
    expect(handleEmailClick).not.toHaveBeenCalled();
    expect(handlePhoneClick).not.toHaveBeenCalled();
  });
  it('Can click the Email button', () => {
    component.find('#email-button').simulate('click');
    expect(handleFbClick).not.toHaveBeenCalled();
    expect(handleEmailClick).toHaveBeenCalled();
    expect(handlePhoneClick).not.toHaveBeenCalled();
  });
  it('Can click the Phone button', () => {
    component.find('#phone-button').simulate('click');
    expect(handleFbClick).not.toHaveBeenCalled();
    expect(handleEmailClick).not.toHaveBeenCalled();
    expect(handlePhoneClick).toHaveBeenCalled();
  });
  it('Displays a phone number', () => {
    expect(component.find('#phone-number').text()).toBe(t('Phone-Number'));
  });
  it('Clicks tabs when tab menu items are clicked', () => {
    component.find(MenuItem).first().simulate('click');
    expect(changeTabs).toHaveBeenCalled();
  });
});
