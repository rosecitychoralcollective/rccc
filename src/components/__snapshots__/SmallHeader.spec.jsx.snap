// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Full header component Renders without crashing 1`] = `"<header class=\\"MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4\\"><div class=\\"MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters\\"><div><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\\"></path></svg></span></button></div><div><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M11.5 2C6.81 2 3 5.81 3 10.5S6.81 19 11.5 19h.5v3c4.86-2.34 8-7 8-11.5C20 5.81 16.19 2 11.5 2zm1 14.5h-2v-2h2v2zm0-3.5h-2c0-3.25 3-3 3-5 0-1.1-.9-2-2-2s-2 .9-2 2h-2c0-2.21 1.79-4 4-4s4 1.79 4 4c0 2.5-3 2.75-3 5z\\"></path></svg></span></button></div></div></header>"`;
